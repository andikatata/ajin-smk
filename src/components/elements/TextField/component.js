import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';

export default class Component extends React.Component {
  render() {
    const {
      autoComplete,
      className,
      disabled,
      input,
      InputProps,
      label,
      meta: { error },
      multiline,
      placeholder,
      readOnly,
      rows,
      type,
      InputLabelProps,
    } = this.props;
    const helperText = error || '';
    const inputProps = {
      readOnly,
      autoComplete,
    };

    return (
      <TextField
        {...input}
        className={className}
        disabled={disabled}
        error={typeof error == 'string'}
        fullWidth
        helperText={helperText}
        label={label}
        margin="normal"
        multiline={multiline}
        placeholder={placeholder}
        inputProps={inputProps}
        InputProps={InputProps}
        rows={rows}
        type={type}
        InputLabelProps={InputLabelProps}
      />
    );
  }
}

Component.propTypes = {
  autoComplete: PropTypes.string,
  classes: PropTypes.object,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  input: PropTypes.object,
  InputProps: PropTypes.object,
  label: PropTypes.string,
  meta: PropTypes.object,
  multiline: PropTypes.bool,
  placeholder: PropTypes.string,
  readOnly: PropTypes.bool,
  rows: PropTypes.number,
  type: PropTypes.string,
  InputLabelProps: PropTypes.object
};