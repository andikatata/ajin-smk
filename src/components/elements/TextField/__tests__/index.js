import React from 'react';
import renderer from 'react-test-renderer';
import TextField from '../index';

describe('TextField', () => {
  it('renders correctly', () => {
    const meta = {
      error: false,
      touched: false,
    };
    const tree = renderer
      .create(<TextField meta={meta} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with helper text', () => {
    const meta = {
      error: 'test',
      touched: true,
    };
    const tree = renderer
      .create(<TextField meta={meta} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});