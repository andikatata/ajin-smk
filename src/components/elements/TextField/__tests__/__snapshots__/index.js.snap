// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders correctly 1`] = `
<div
  className="MuiFormControl-root-2 MuiFormControl-marginNormal-3 MuiFormControl-fullWidth-5"
>
  <div
    className="MuiInput-root-6 MuiInput-fullWidth-13 MuiInput-formControl-7 MuiInput-underline-10"
  >
    <input
      aria-invalid={false}
      className="MuiInput-input-14"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
  
</div>
`;

exports[`TextField renders with helper text 1`] = `
<div
  className="MuiFormControl-root-2 MuiFormControl-marginNormal-3 MuiFormControl-fullWidth-5"
>
  <div
    className="MuiInput-root-6 MuiInput-error-11 MuiInput-fullWidth-13 MuiInput-formControl-7 MuiInput-underline-10"
  >
    <input
      aria-invalid={true}
      className="MuiInput-input-14"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
  </div>
  <p
    className="MuiFormHelperText-root-19 MuiFormHelperText-error-20"
  >
    test
  </p>
</div>
`;
